lib/yeti_exception/error.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module YetiException
+
+  # Enhanced exception class, with some features added beyond those in
+  # StandardError:
+  #
+  # - The class in which the exception was raised.  This can be useful for
+  #   logging, since the backtrace is much more verbose.
+  #
+  # - A details hash, containing arbitrary key/value pairs related to the
+  #   exception.
+  #
+  # - A transient flag, defaulting to true, indicating whether the exception
+  #   should be considered transient or final.  This can be used when rescuing
+  #   the exception to determine whether the operation should be retried.
+  #
+  # - An HTTP status, defaulting to 500.  This can be used to generate the
+  #   response payload if the exception is rescued as part of an HTTP request.
+  #
+  # - An optional static message, definable by subclasses, to be included
+  #   automatically in the details hash.  This removes the need to pass this
+  #   message in explicitly during initialization.
+
+  class Error < StandardError
+
+    # @return [Class] Class in which the exception occurred
+    attr_reader :klass
+    # @return [Hash] Arbitrary details about the exception
+    attr_reader :details
+    # @return [TrueClass, FalseClass] Whether the error is transient
+    attr_reader :transient
+    # @return [Integer] Corresponding HTTP status
+    attr_reader :http_status
+
+    # @param klass [Class] Class in which the exception occurred
+    # @param details [Hash] Arbitrary details about the exception
+    # @param transient [TrueClass, FalseClass] Whether the error is transient
+    # @param http_status [Integer] Corresponding HTTP status
+    def initialize(klass, details, transient = true, http_status = 500)
```

Comment:
[nit] Are most errors transient? I would expect it to be the opposite.

Reply:
I'm not sure, but given that the default behavior of tools like Sidekiq, Sneakers, etc. is to retry a job on an exception, I decided to make transient the default.